src/plugins/plugin-http.ts

```diff
@@ -111,6 +121,24 @@ function makeRequestTrace(
     span.addLabel(api.labels.HTTP_METHOD_LABEL_KEY, method);
     span.addLabel(api.labels.HTTP_URL_LABEL_KEY, uri);
 
+    // If outgoing request headers contain the "Expect" header, the returned
+    // ClientRequest will throw an error if any new headers are added. For this
+    // reason, only in this scenario, we opt to clone the options object to
+    // inject the trace context header instead of using ClientRequest#setHeader.
+    // (We don't do this generally because cloning the options object is an
+    // expensive operation.)
+    let traceHeaderPreinjected = false;
+    if (hasExpectHeader(options)) {
+      traceHeaderPreinjected = true;
+      // "Clone" the options object -- but don't deep-clone anything except for
+      // headers.
+      options = Object.assign({}, options);
```

Comment:
Nit. `options = {...options};` would be simpler.

Reply:
I think `...` syntax isn't available in Node 4?

Reply:
I believe TS transpiles it to what you wrote here.